presto-cassandra/src/main/java/com/facebook/presto/cassandra/util/SslContextProvider.java

```diff
@@ -0,0 +1,168 @@
+            Optional<String> truststorePassword)
+    {
+        this.keystorePath = keystorePath;
+        this.keystorePassword = keystorePassword;
+        this.truststorePath = truststorePath;
+        this.truststorePassword = truststorePassword;
+    }
+
+    public Optional<SSLContext> buildSslContext()
+    {
+        if (!keystorePath.isPresent() && !truststorePath.isPresent()) {
+            return Optional.empty();
+        }
+        try {
+            // load KeyStore if configured and get KeyManagers
+            KeyStore keystore = null;
+            KeyManager[] keyManagers = null;
+            if (keystorePath.isPresent()) {
+                char[] keyManagerPassword;
+                try {
+                    // attempt to read the key store as a PEM file
+                    keystore = loadKeyStore(keystorePath.get(), keystorePath.get(), keystorePassword);
+                    // for PEM encoded keys, the password is used to decrypt the specific key (and does not
+                    // protect the keystore itself)
+                    keyManagerPassword = new char[0];
+                }
+                catch (IOException | GeneralSecurityException ignored) {
+                    keyManagerPassword = keystorePassword.map(String::toCharArray).orElse(null);
+                    keystore = KeyStore.getInstance(KeyStore.getDefaultType());
+                    try (InputStream in = new FileInputStream(keystorePath.get())) {
+                        keystore.load(in, keyManagerPassword);
+                    }
+                }
+                validateCertificates(keystore);
+                KeyManagerFactory keyManagerFactory =
+                        KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+                keyManagerFactory.init(keystore, keyManagerPassword);
+                keyManagers = keyManagerFactory.getKeyManagers();
+            }
+            // load TrustStore if configured, otherwise use KeyStore
+            KeyStore truststore = keystore;
+            if (truststorePath.isPresent()) {
+                truststore = loadTrustStore(truststorePath.get(), truststorePassword);
+            }
+
+            // create TrustManagerFactory
+            TrustManagerFactory trustManagerFactory =
+                    TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            trustManagerFactory.init(truststore);
+
+            // get X509TrustManager
+            TrustManager[] trustManagers = trustManagerFactory.getTrustManagers();
+            if ((trustManagers.length != 1) || !(trustManagers[0] instanceof X509TrustManager)) {
+                throw new RuntimeException("Unexpected default trust managers:" + Arrays.toString(trustManagers));
+            }
+
+            X509TrustManager trustManager = (X509TrustManager) trustManagers[0];
+            // create SSLContext
+            SSLContext result = SSLContext.getInstance("SSL");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

An old version of SSL/TLS is enabled. We recommend that you enable TLS version 1.2, 1.3, or later. [Learn more](https://cheatsheetseries.owasp.org/cheatsheets/TLS_Cipher_String_Cheat_Sheet.html)